### PR TITLE
955 clear out uses of addAction from tests

### PR DESCRIPTION
**Pull request by cz4rs**
remove remaining usage of addAction from:
`tests/unit/pipe/test_callback_bcast_collection.extended.cc`
`tests/unit/location/test_location_common.h`

### CODE REVIEW

**Comment by lifflander** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ tests/unit/pipe/test_callback_bcast_collection.extended.cc  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/lifflander/vt/pullRequest?prid=6032395&bid=19769214)

File: tests/unit/pipe/test_callback_bcast_collection.extended.cc [hunk 1]
```diff
@@ -130,18 +128,26 @@ static void cb3(DataMsg* msg, TestCol* col) {
...
+    if (this_node == 0) {
+      auto cb = theCB()->makeBcast<TestCol,DataMsg,&TestCol::cb1>(proxy);
+      auto nmsg = makeMessage<DataMsg>(8,9,10);
+      cb.send(nmsg.get());
+    }
+  });
+
+  runInEpochCollective([&]{
+    if (this_node == 0) {
+      for (auto i = 0; i < 32; i++) {
```
Comment (JacobDomagala) [inline tests/unit/pipe/test_callback_bcast_collection.extended.cc:144]: No need for that for loop ;)

Comment (cz4rs) [reply tests/unit/pipe/test_callback_bcast_collection.extended.cc]: removed :+1:

**Comment by lifflander** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- tests/unit/pipe/test_callback_bcast_collection.extended.cc  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/lifflander/vt/pullRequest?prid=6032395&bid=19769214)

File: tests/unit/pipe/test_callback_bcast_collection.extended.cc [hunk 2]
```diff
@@ -130,18 +128,24 @@ static void cb3(DataMsg* msg, TestCol* col) {
...
-
-    theTerm()->addAction([=]{
-      proxy.destroy();
-    });
-  }
+  runInEpochCollective([&]{
+    if (this_node == 0) {
+      auto cb = theCB()->makeBcast<TestCol,DataMsg,&TestCol::check>(proxy);
+      auto nmsg = makeMessage<DataMsg>();
+      cb.send(nmsg.get());
```
Comment (PhilMiller) [inline tests/unit/pipe/test_callback_bcast_collection.extended.cc:146]: This test is supposed to verify the callback code. The ultimate assertions that it worked correctly are in `TestCol::check`. So, that should be reached by a *non* callback path - i.e. a direct broadcast

Comment (PhilMiller) [reply tests/unit/pipe/test_callback_bcast_collection.extended.cc:146]: And the same applies below

Comment (PhilMiller) [reply tests/unit/pipe/test_callback_bcast_collection.extended.cc:146]: To understand why I say this, imagine a defect in callbacks that caused them not to be delivered at all. Everything would reach termination, with no assertion failures, and the test would pass.

**Comment by lifflander** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- tests/unit/pipe/test_callback_bcast_collection.extended.cc  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/lifflander/vt/pullRequest?prid=6032395&bid=19769214)

**Comment by lifflander** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- tests/unit/pipe/test_callback_bcast_collection.extended.cc  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/lifflander/vt/pullRequest?prid=6032395&bid=19769214)

**Comment by lifflander** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- tests/unit/pipe/test_callback_bcast_collection.extended.cc  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/lifflander/vt/pullRequest?prid=6032395&bid=19769214)

File: tests/unit/location/test_location_common.h [hunk 3]
```diff
@@ -147,65 +147,50 @@ void verifyCacheConsistency(
...
-        }
-      } else /* my_node == home */ {
-        // The entity should be registered in the cache of the home node,
-        // regardless of the protocol (eager or not)
-        EXPECT_TRUE(isCached(entity));
+        // route entity message
+        vt::theLocMan()->virtual_loc->routeMsg<MsgT>(entity, home, msg, serialize);
       }
-      finished = true;
     });
```
Comment (cz4rs) [inline tests/unit/location/test_location_common.h]: the checks should execute after the action now

**Comment by cz4rs** [comment]
rebased on top of develop

@lifflander @PhilMiller 
2d6722278e8f13346e35610a6fd3650ae36fed64 should fix the issue introduced in `test_location_common.h`

**Comment by lifflander** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- tests/unit/pipe/test_callback_bcast_collection.extended.cc  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/lifflander/vt/pullRequest?prid=6032395&bid=19769214)

**Comment by lifflander** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- tests/unit/pipe/test_callback_bcast_collection.extended.cc  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/lifflander/vt/pullRequest?prid=6032395&bid=19769214)

**Comment by PhilMiller** [comment]
The code looks good to me now. @lifflander could you double-check the location manager test change?

**Comment by lifflander** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- tests/unit/pipe/test_callback_bcast_collection.extended.cc  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/lifflander/vt/pullRequest?prid=6032395&bid=19769214)

**Comment by lifflander** [comment]
Looks good to me, let's get it merged

**Pull request merged**